lizard_auth_server/views.py

```diff
@@ -304,3 +324,190 @@ def profile(self):
         if not self._profile:
             self._profile = self.invitation.user.get_profile()
         return self._profile
+
+
+class JWTView(FormView):
+    """A view that provides a JSON Web Token after successful authentication.
+
+    The query string may have a `next` parameter used for redirection. If
+    present, it must be a fully qualified URL starting with http(s). The
+    token will be attached to the query string as an extra parameter:
+    <next>?foo=bar&access_token=xxxxx.yyyyy.zzzzz&...
+
+    If a `next` parameter is absent, a JSON response is returned:
+
+    {"access_token": "xxxxx.yyyyy.zzzzz"}
+
+    """
+    template_name = 'lizard_auth_server/login.html'
+    form_class = forms.JWTAuthenticationForm
+
+    @staticmethod
+    def is_url(url):
+        """Validate a URL.
+
+        The current validator accepts only absolute URLs, which makes
+        sense as the token is to be generated for another portal.
+
+        Args:
+            url (str): an absolute, i.e. fully qualified, URL.
+
+        Returns:
+            bool: True if valid, False otherwise.
+
+        """
+        val = URLValidator()
+        try:
+            val(url)
+            return True
+        except ValidationError:
+            return False
+
+    @staticmethod
+    def is_portal(sso_key):
+        """Validate a portal.
+
+        Args:
+            sso_key (str): a public key identifying.
+
+        Returns:
+            bool: True if the portal exists, False otherwise.
+
+        """
+        return Portal.objects.filter(sso_key=sso_key).exists()
+
+    def get(self, request, *args, **kwargs):
```

Comment:
Bij deze en de `post()` begin ik te twijfelen. Het is een methode die aan de ene kant json terug kan geven en aan de andere kant ook een html invulformulier. Is dat handig? 

De gebruiker van deze API moet dan goed oppassen bij het gebruik. Is het altijd ook een webbrowser? Of kan een app ook inloggen met een apart app wachtwoord? Die snapt geen html formulier.

En "get" gebruiken voor het snel verversen van een token: is dat (qua foutmelding/afhandeling) voor de API-gebruiker niet handiger als het een losse `/jwt/refresh-token/` url is?